internal/databroker/server.go

```diff
@@ -222,30 +237,21 @@ func (srv *Server) Sync(req *databroker.SyncRequest, stream databroker.DataBroke
 		return err
 	}
 
-	ch := srv.onchange.Bind()
-	defer srv.onchange.Unbind(ch)
-	for {
-		updated, _ := db.List(context.Background(), recordVersion)
-		if len(updated) > 0 {
-			sort.Slice(updated, func(i, j int) bool {
-				return updated[i].Version < updated[j].Version
-			})
-			recordVersion = updated[len(updated)-1].Version
-			err := stream.Send(&databroker.SyncResponse{
-				ServerVersion: srv.version,
-				Records:       updated,
-			})
-			if err != nil {
-				return err
-			}
-		}
+	ctx := stream.Context()
 
-		select {
-		case <-stream.Context().Done():
-			return stream.Context().Err()
-		case <-ch:
+	// Do first sync, so we won't missed anything.
+	if err := srv.doSync(ctx, recordVersion, db, stream); err != nil {
+		return err
+	}
+
+	ch := db.Sync(ctx)
+
+	for range ch {
```

Comment:
This will block. We need to use a select with `ctx.Done()`

Reply:
Not sure we need. `db.Sync(ctx)` is called with `ctx` is `stream.Context()`, when ctx .Done(), `ch` will be closed inside `db.Sync(ctx)`, so the for loop will be terminated.